packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -0,0 +1,195 @@
+import type { Plugin } from 'vite';
+import { readdir, writeFile } from 'node:fs/promises';
+import { existsSync, readFileSync } from 'node:fs';
+import path from 'node:path';
+import { SRC_ENTRIES, SRC_PAGES } from '../constants.js';
+import { joinPath } from '../utils/path.js';
+
+const srcToName = (src: string) => {
+  const split = src
+    .split('/')
+    .map((part) => (part[0]!.toUpperCase() + part.slice(1)).replace('-', '_'));
+
+  if (src.endsWith('_layout.tsx')) {
```

Comment:
This will confuse with `/pages/foo_layout.tsx`.

Reply:
How about this?
```suggestion
  if (split.at(-1) === '_layout.tsx') {
```

Reply:
good point, fixed!